### DISCUSSION TITLE
Display icon of current app in bar.

### DISCUSSION THREAD

**Discussion by ajoino** [qtile/qtile#4929] [Q&A] [answered]
I would like to display the currently focused application's icon to the left of the `WindowName` widget, or inside of it as part of the text. Is this possible to do on Wayland?

I've looked around and googled and the closest thing I can find is using `TaskList` but I just want the single icon.

**Comment by elParaguayo** [comment, marked as answer]
You'd have to create a custom widget to do it.

**Reply by ajoino** [reply]
Thanks for the quick answer! Had basically figured as much but wanted to make sure I wasn't overlooking anything obvious.

**Reply by elParaguayo** [reply]
Sorry for the short answer. Did it on my phone.

Longer answer is that you'd need to lift the code for getting the app icon (eg from Tasklist widget) and then just have your widget render it. The widget would use hooks to know when focus had changed to a different app.

If you give it a go and get stuck, please do come and ask for help here.

This page in the docs may also help too: https://docs.qtile.org/en/stable/manual/howto/widget.html

**Reply by ajoino** [reply]
Doesn't look too hard to do, I might give it a go this weekend.
